.design/requirements-575.md

```diff
@@ -0,0 +1,269 @@
+# Jakarta WebSocket Whiteboard
+
+This documents looks into the requirements for a Jakarta WebSocket whiteboard specification.
+
+## Introduction
+
+The OSGi Compendium R6 introduced the HTTP Whiteboard specification, upgraded in R8.1 to the Whiteboard specification for Jakarta Servlet.
+It provides a light and convenient way of using servlets in a an OSGi environment through the use of the Whiteboard Pattern.
+
+WebSocket is communication protocol designed to work over the HTTP protocol.
```

Comment:
This is wrong. see for example: https://en.wikipedia.org/wiki/WebSocket

> WebSocket is a computer [communications protocol](https://en.wikipedia.org/wiki/Communications_protocol), providing a [simultaneous two-way](https://en.wikipedia.org/wiki/Full-duplex) communication channel over a single [Transmission Control Protocol](https://en.wikipedia.org/wiki/Transmission_Control_Protocol) (TCP) connection.

and further down

> WebSocket is distinct from [HTTP](https://en.wikipedia.org/wiki/HTTP) used to serve most webpages. Although they are different, [RFC](https://en.wikipedia.org/wiki/RFC_(identifier)) [6455](https://datatracker.ietf.org/doc/html/rfc6455) states that WebSocket "is designed to work over HTTP ports 443 and 80 as well as to support HTTP proxies and intermediaries", thus making it compatible with HTTP. To achieve compatibility, the WebSocket [handshake](https://en.wikipedia.org/wiki/Handshake_(computing)) uses the [HTTP Upgrade header](https://en.wikipedia.org/wiki/HTTP/1.1_Upgrade_header)[[3]](https://en.wikipedia.org/wiki/WebSocket#cite_note-3) to change from the HTTP protocol to the WebSocket protocol.

So HTTP and Websocket are distinct protocols and HTTP supports switching of protocols (not only Websocket!).

The protocoll for websockets is `ws://` (or `wss://` for ssl), while HTTP uses `http://` and `https://`